addons/mrp/models/stock_warehouse.py

```diff
@@ -274,6 +274,14 @@ def _get_picking_type_update_values(self):
         })
         return data
 
+    def _create_missing_locations(self, vals):
+        super()._create_missing_locations(vals)
+        for warehouse in self:
```

Comment:
loop on warehouse's company instead of warehouse themselves to avoid creating too many locations